projects/vdk-core/src/vdk/internal/core/config.py

```diff
@@ -188,6 +203,7 @@ def add(
         """
         key = _normalize_config_key(key)
         self.__config_key_to_default_value[key] = default_value
+        self.__config_key_to_can_override[key] = True
```

Comment:
what is the rule? new config properties can be overridden? 
Can you give examples of properties that can be overridden and ones that can't

Reply:
You can override any config key-value pair with a secret, as long as you have a secret with the same key.

Reply:
ok if you can override anyone then why not remove this? and we can just override everyone?